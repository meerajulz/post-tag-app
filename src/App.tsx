import theme from './styles/theme';
import InputMessage from './components/input-message/input-message.component';
import { ThemeProvider } from 'styled-components';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <main className="container mx-auto p-4">
        <header className="App-header">
          <h1>Medical Report Auto-Complete</h1>
        </header>
        <div className="container mx-auto p-4">
          <InputMessage />
        </div>
      </main>
    </ThemeProvider>
  );
}

export default App;
