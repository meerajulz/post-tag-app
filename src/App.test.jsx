import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('App Component', () => {
  test('renders without crashing', () => {
    render(<App />);
    expect(
      screen.getByRole('heading', { name: /Medical Report Auto-Complete/i })
    ).toBeInTheDocument();
    expect(
      screen.getByText(/Medical Report Auto-Complete/i)
    ).toBeInTheDocument();
  });

  test('includes InputMessage component', () => {
    render(<App />);
    expect(screen.getByTestId('message-container')).toBeInTheDocument(); // Ensure InputMessage has a 'data-testid="message-container"'
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
