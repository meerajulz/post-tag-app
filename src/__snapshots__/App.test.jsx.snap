// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="container mx-auto p-4"
  >
    <header
      class="App-header"
    >
      <h1>
        Medical Report Auto-Complete
      </h1>
    </header>
    <div
      class="container mx-auto p-4"
    >
      <div
        class="sc-blHHSb ktMXbQ"
        data-testid="message-container"
      >
        <div
          class="sc-gtLWhw gGrQjy"
        >
          <span>
             
          </span>
        </div>
        <textarea
          class="sc-egkSDF yhvOw"
          placeholder="Whatâ€™s happening?"
        />
      </div>
    </div>
  </main>
</DocumentFragment>
`;
